scipy/ndimage/tests/test_morphology.py

```diff
@@ -2521,6 +2520,8 @@ def test_white_tophat01(self, xp):
         tmp = ndimage.grey_opening(array, footprint=footprint,
                                    structure=structure)
         expected = array - tmp
+        # array created by xp.zeros is non-writeable for dask
+        # and jax
```

Comment:
Not true. All arrays are non-writeabla for jax. All arrays are writeable for dask.

Reply:
r.e. writeability, I am not referring to the mutability of the original dask array, but the numpy array created from the dask array with np.asarray.

MRE
```
import dask.array as da
a = da.zeros(10)
a_np = np.asarray(a)
a_np[1] = 10
Traceback (most recent call last):
  File "<python-input-4>", line 1, in <module>
    a_np[1] = 10
    ~~~~^^^
ValueError: assignment destination is read-only
```

I'll remove the jax comment (and I'll raise an issue on scipy to disallow the ``output`` keyword for jax since it looks like there's no way to make that work).

Reply:
The whole idea of having an `output=` parameter makes no sense if you are going to convert it to numpy.

This makes sense:
```python
def f(x, output):
    output[:] = x * 2
```
This does not:
```
def f(x, output):
    x = np.asarray(x)
    output = np.asarray(output)
    output[:] = x * 2
```
In the second example, even without the broadcasting issue you're facing, the `output` item that the user is holding is still completely blank *unless* np.asarray is a view of the original object. which, for dask, can never be.

Reply:
Note that `np.asarray(output, copy=False)` will not crash, while it should.
This is a limitation of the `__array__` interface, that doesn't inform Dask that the user expects strictly a memory view of the data, which is impossible.

Reply:
Yeah, makes sense.

I'll put in a separate PR to the main scipy repo disabling the output kwarg for jax (to keep the diff down here).

Reply:
FYI https://github.com/dask/dask/pull/11697